src/dash-playlist-loader.js

```diff
@@ -204,7 +206,44 @@ export const requestSidx_ = (sidxRange, playlist, xhr, options, finishProcessing
     headers: segmentXhrHeaders(sidxInfo)
   });
 
-  return xhr(sidxRequestOptions, finishProcessingFn);
+  return containerRequest(sidxInfo.uri, xhr, (err, request, container, bytes) => {
+    if (err) {
+      return finishProcessingFn(err, request);
+    }
+
+    if (!container || container !== 'mp4') {
+      this.error = {
+        status: request.status,
+        message: `Unsupported ${container || 'unknown'} container type for sidx segment at URL: ${request.uri}`,
+        response: request.response,
+        playlist,
+        internal: true,
```

Comment:
Are we ok with this being an internal error? Seems like a feature we lack rather than an actual error to me, but if we want to warn in the console that could be fine.

Reply:
I think we can just leave it as is for now, we probably want to rethink how errors are handled to be able to better allow for people to handle things as they want.